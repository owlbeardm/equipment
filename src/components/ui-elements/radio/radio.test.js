import React from 'react'
import { shallow } from 'enzyme'

import Radio from './radio'
import RadioGroup from './radio-group'

const defaultRadioProps = { input: { value: '', onChange: () => { } }, label: 'the radio', value: 'the-radio' }
const childElements = [
  shallow(<Radio {...defaultRadioProps} />),
  shallow(<Radio {...defaultRadioProps} />)
]
const defaultGroupProps = { input: {}, children: childElements }

describe('Radio component testing', () => {
  test('renders without error', () => {
    const wrapper = shallow(<Radio {...defaultRadioProps} />)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('RadioGroup component testing', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<RadioGroup {...defaultGroupProps} />)
  })

  test('renders without error', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders all children elements', () => {
    const childrenNumber = wrapper.find('input[type="radio"]')
    expect(childrenNumber.length).toBe(childElements.length)
  })
})
