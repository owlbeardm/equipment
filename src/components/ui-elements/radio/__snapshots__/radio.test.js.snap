// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component testing renders without error 1`] = `"<div class=\\"form-check form-check-radio\\"><label class=\\"form-check-label text-dark\\"><input type=\\"radio\\" value=\\"the-radio\\" id=\\"the-radio\\" class=\\"form-check-input\\"/>the radio<span class=\\"circle\\"><span class=\\"check\\"></span></span></label></div>"`;

exports[`RadioGroup component testing renders without error 1`] = `"<div class=\\"form-group\\"><div class=\\"form-check form-check-radio\\" input=\\"[object Object]\\"><label class=\\"form-check-label text-dark\\"><input type=\\"radio\\" value=\\"the-radio\\" id=\\"the-radio\\" class=\\"form-check-input\\"/>the radio<span class=\\"circle\\"><span class=\\"check\\"></span></span></label></div><div class=\\"form-check form-check-radio\\" input=\\"[object Object]\\"><label class=\\"form-check-label text-dark\\"><input type=\\"radio\\" value=\\"the-radio\\" id=\\"the-radio\\" class=\\"form-check-input\\"/>the radio<span class=\\"circle\\"><span class=\\"check\\"></span></span></label></div></div>"`;
